src/mqtt.js

```diff
@@ -622,10 +623,10 @@ function obfuscateCoordinate(coordinate) {
     if (coordinate.includes(".")) {
         const parts = coordinate.split(".");
         let precisionPart = parts[1];
-        // This will catch anything 1.1km or more precise, or two decimal places
-        if (parts[1].length > 1) {
+        // Takes the configurable precision, and randomizes the last point of precision
+        if (parts[1].length >= obfuscatePrecision) {
             // Take only the first digit
-            precisionPart = precisionPart.substring(0, 1);
+            precisionPart = precisionPart.substring(0, obfuscatePrecision - 1);
             // Add a random digit for the 1.1km precision point
             precisionPart += Math.floor(Math.random() * 10);
```

Comment:
I think this line doesn't take into account the precision selected. It looks like it always adds a digit to the MSD of the precisionPart.

Reply:
Did you run the fiddle? Example... If `obfuscatePrecision` is 2. This will include 2 decimals of precision but the last decimal is randomized. So the new coordinate is as precise as configured (2 decimals) but technically less accurate (intentionally).

43.2342342343 becomes 43.29 where the 9 is a random digit
-81.12 becomes -81.19 where 9 is a random digit
-81.3241 becomes -81.35 where 5 is a random digit.

Reply:
Yeah you are right. I didn't realize precisionPart is a string. JS is weird.